src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    static final String VALUE_SUPPLY = "supply";
+    static final String VALUE_BUY = "buy";
+    static final String VALUE_RESULT = "result";
+    static final int ZERO_INDEX = 0;
+    static final int FIRST_INDEX = 1;
+    static final String SEPARATOR = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            StringBuilder dataFromFile = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                dataFromFile.append(line).append(System.lineSeparator());
+            }
+            String report = createReport(dataFromFile.toString());
+            writeReportToFile(report, toFileName);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fromFileName, e);
+        }
```

Comment:
please create a separate method for read operation as well. You should call three methods inside getStatistics with the next purposes: read, generate report, write